vllm/core/scheduler.py

```diff
@@ -1019,6 +1039,8 @@ def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
             self.block_manager.mark_blocks_as_computed(
                 scheduled_seq_group.seq_group)
 
+        seq_group_metadata_cache.reset()
+
```

Comment:
the cache is a global object, but we can have several schedulers (in pipeline parallel). this means one scheduler might clean the cache when another scheduler is using the objects from the cache, right?

maybe we should make the cache local to each scheduler.

Reply:
Thanks for catching this important issue, I will make the cache local per scheduler instance.